### DISCUSSION TITLE
Forward OCPP to National Grid Rebate program

### DISCUSSION THREAD

**Discussion by amaisano** [lbbrhzn/ocpp#1234] [Q&A]
Is it possible to either:

A) Have my Wallbox connect to MULTIPLE OCPP controllers or

B) Have this OCPP integration "forward" its charger data to my state's rebate program controller (ev.energy ChargeSmart MA)?

The main issue I have is once I set my Wallbox to use my State's OCPP provider, I lose ALL control over my $600 Wallbox charger. I cannot control charging ON or charging OFF. I cannot change most settings. It stinks.

**Comment by lbbrhzn** [comment]
A) This depends on your charger. Some charges do support multiple servers. For instance, the Wallbox Plus can connect simultaneously to the proprietary Wallbox backend, and to a generic OCPP server. It cannot connect to two generic OCPP servers though, and you lose control over part of the settings.

B) The integration does not support that.

 I think you would need to create some kind of OCPP proxy service, that allows you to split an OCPP connection.  When one of the servers sends a request, the proxy will forward it to the charger, and the response is sent back to the appropriate server.  When the charger sends a notification, both servers will get it, and both will respond. You can only send back one of these responses to the charger, so you need to decide which one, maybe a preferred one. The difficult part is how to resolve state conflicts: When the two servers are trying to configure the charger into a different state,  you will likely end up with a flood of conflicting control messages. So probably you need a virtual charger that maintains a separate state from the real one, and only copies part of the reporting data of the real charger.

**Reply by amaisano** [reply]
Thanks. Well if anyone has any idea on how to do this I'd love to hear it

I have a Wallbox Pulsar Plus 40a. It seems to only have one URL field for OCPP though, and it's set to `wss://ocpp.ev.energy`

**Comment by drc38** [comment]
Fyi https://ocpp-spec.org/docs/architecture/supported-topologies/ provides details of how the 2.0.1 ocpp architecture is intended to be used/connected.
